
## coGSEA Manual

### Introduction

**coGSEA** (**co**mparative **G**ene **S**et **E**nrichment **A**nalysis) is a tool developped to help you compare up to 12 different methods of GSEA.
In the current version, those 12 methods include :
- [camera](http://www.bioconductor.org/packages/release/bioc/html/limma.html)
- [gage](https://bioconductor.org/packages/release/bioc/html/gage.html)
- [globaltest](https://bioconductor.org/packages/release/bioc/html/globaltest.html)
- [gsva](https://bioconductor.org/packages/release/bioc/html/GSVA.html)
- [ssgsea](https://bioconductor.org/packages/release/bioc/html/GSVA.html)
- [zscore](https://bioconductor.org/packages/release/bioc/html/GSVA.html)
- [plage](https://bioconductor.org/packages/release/bioc/html/GSVA.html)
- [ora](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3134237/)
- [padog](http://bioconductor.org/packages/release/bioc/html/PADOG.html)
- [roast](http://www.bioconductor.org/packages/release/bioc/html/limma.html)
- [safe](http://bioconductor.org/packages/release/bioc/html/safe.html)
- [setrank](https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-017-1571-6)

#### Disclaimer :
This tool is largely inspired by the [eGSEA](http://bioconductor.org/packages/release/bioc/html/EGSEA.html) R package (and contains some of its code).
Note that because of optimization of computation time, the SetRank method is not included in this GUI version of coGSEA. To include, you can refer to the R package [coGSEA_SetRank](https://gitlab.pasteur.fr/mborry/coGSEA_SetRank)

### Methods
After running up to 12 methods :
- For each method, *coGSEA* ranks the gene sets on the raw p.value
- *coGSEA* adjust the p.values returned each GSEA method using the pvalue adjustment method of your choice.
- *coGSEA* combines the adjusted p.values of each GSEA method using the p.value combination method of your choice.
- *coGSEA* does a clustering, correlation, and PCA analysis of the methods, using the ranks of the gene sets in each method.
- *coGSEA* does an ensemblist analysis on the gene sets with an adjusted p.value, for each method, lower than the user defined $\alpha$ treshold.
- Last but not least, *coGSEA* performs a summarization of the data using both the average logFC (log Fold Change) and p.value of each geneset, accros the different methods.

### Input
- **Elist .rds file** : a .rds file containing an Elist object. This file can be generated by saving your Elist object using the `saveRDS()` function.   
Your Elist object must be either a *limma-voom* object in case of RNAseq data, or have a estimated dispersion in case of MicroArray data. Dispersion estimates can be performed using the `estimateDisp()` function from the *edgeR* package.
- **ENTREZGenesIds** : the name of the column containing all the ENTREZ identifiers of the genes in your data. This information is located in the "genes" matrix of the Elist object. Hence, you can access this information by running `colnames(Elist$genes)`.
This value is often named *ENTREZID* or *ENTREZ* or *FeatureID*.
- **Contrast .rds file** : a .rds file containing the contrast matrix for your experiment. For some help to make a contrast matrix, you can check the vignette of the limma package.
Once the contrast matrix is created, you can save it to rds file using the `saveRDS()` function.
- **MSigDB geneset** : A geneset collection from the [MSigDB](http://software.broadinstitute.org/gsea/msigdb) database. Currently only two are supported : H (Hallmark) and a subset of C2(Kegg and Reactome). More will maybe be added in the future.
- **Specie** : The organism from which the data were extracted. Currently only *Homo sapiens* and *Mus musculus* are supported.
- **GSEA Methods** : between 1 and 12 methods to select from the methods listed in the introduction.   
Note : including SetRank will significantly increase the calculation time.
- **Alpha error treshold** : the null hypothesis will be rejected if the p.value/adjusted p.value/combined p.value is lower than alpha.
- **p.value adjustment method** : the method to adjust the p.value of the different genesets in one GSEA method, to select from the following :
	- holm
	- hochberg
	- hommel
	- bonferroni
	- BH (Benjamini Hochberg) - selected by default.
	- BY (Benjamini Yekutieli)
	- fdr (False Discovery rate)
	- none
- **p.value combination method** : the method to combine all the p.values of one geneset accros the different GSEA methods, to select from the following :
	- sumz (sum z method)
	- votep (vote counting method)
	- minimump (Wilkinson's method)
	- sumlog (Fisher's method) - selected by default
	- sump (sum p method)
	- logitp (logit method)
	- meanp (mean p method)
	- maximump (Wilkinson's method)
- **Minimum intersection size for SnailPlot** : Graphical Parameter to select the minimum size of an intersection of genesets between different methods
